### ISSUE TITLE
Werkzeug update to 3.0.0 breaks master branch builds

### ISSUE THREAD

**Issue by aptalca** [janeczku/calibre-web#2904]
**Describe the bug/problem**
When calibre-web is built from master branch, it is no longer starting since Werkzeug got updated to 3.0.0

**To Reproduce**
1. Install calibre-web from master branch
2. Start it

**Logfile**
```
Traceback (most recent call last):
  File "/app/calibre-web/cps.py", line 28, in <module>
    from cps.main import main
  File "/app/calibre-web/cps/__init__.py", line 29, in <module>
    from .MyLoginManager import MyLoginManager
  File "/app/calibre-web/cps/MyLoginManager.py", line 24, in <module>
    from flask_login import LoginManager, confirm_login
  File "/usr/local/lib/python3.10/dist-packages/flask_login/__init__.py", line 12, in <module>
    from .login_manager import LoginManager
  File "/usr/local/lib/python3.10/dist-packages/flask_login/login_manager.py", line 33, in <module>
    from .utils import _create_identifier
  File "/usr/local/lib/python3.10/dist-packages/flask_login/utils.py", line 14, in <module>
    from werkzeug.urls import url_decode
ImportError: cannot import name 'url_decode' from 'werkzeug.urls' (/usr/local/lib/python3.10/dist-packages/werkzeug/urls.py)
```

**Expected behavior**
Clean start

**Screenshots**
N/A

**Environment (please complete the following information):**
 - OS: Docker on Ubuntu 22.04
 - Python version: 3.10
 - Calibre-Web version: master HEAD
 - Docker container: LinuxServer nightly tag
 - Special Hardware: x86_64
 - Browser: N/A

**Additional context**
I am one of the team members of Linuxserver.io
Our nightly tag, which builds from the master branch started failing a few days ago, when Werkzeug got updated to 3.0.0
Our last commit before broken builds: https://github.com/linuxserver/docker-calibre-web/commit/8145bd6c9b065767c06198f083f85e764ff45b99
CI test results with full logs: https://ci-tests.linuxserver.io/linuxserver/calibre-web/nightly-198bff92-ls113/index.html
Our repo for the docker container: https://github.com/linuxserver/docker-calibre-web/tree/nightly

Thanks

**Comment by mohammed90** [comment]
This shouldn't be considered a bug in calibre-web. The release of Werkzeug version 3.0 implies it's a breaking release by the major version bump, for the same reason Flask released version 3.0 from the 2.x.y. Using Werkzeug 3.0 while the calibre-web's requirement lists Flask 2.x.y. doesn't typically imply the assumption of a successful build.

**Comment by OzzieIsaacs** [comment]
Flask has a minimum version for Werkzeug, but they don't specify a maximum version. In the past it happened several times that the Werkzeug api was changed and was incompatible. From my point of view, flask should limit the maximum version of werkzeug in their release, what they currently not do, so calibre-web has the problem. (The same problem occurred right before the latest release 0.6.20).  Flask 3.0.0 currently can't be used, as flask-login also relies on the "old" api of werkzeug, and therefore fails with new flask versions. The newest master commit of flask-login has fixed this, but till this isn't release we have to live with what we have now

I included now a maximum version of Werkzeug in the requirements file, this should work for new installations.
